tests/Sinch.Tests/Core/HttpTests.cs

```diff
@@ -191,5 +196,99 @@ public async Task AddOwnHeaders()
 
             _httpMessageHandlerMock.VerifyNoOutstandingExpectation();
         }
+
+        [Fact]
+        public async Task UnauthorizedIfExpiredHeaderIsNotPresent()
+        {
+            _tokenManagerMock.GetAuthToken(Arg.Any<bool>())
+                .Returns("first_token");
+
+            var uri = new Uri("http://sinch.com/items");
+
+            // first token expires
+            _httpMessageHandlerMock.Expect(HttpMethod.Get, uri.ToString())
+                .WithHeaders("Authorization", "Bearer first_token")
+                .Respond(HttpStatusCode.Unauthorized);
+
+            var httpClient = new HttpClient(_httpMessageHandlerMock);
+            var http = new Http(_tokenManagerMock, httpClient, null, new SnakeCaseNamingPolicy());
+            Func<Task<EmptyResponse>> op1 = () => http.Send<EmptyResponse>(uri, HttpMethod.Get);
+
+            await op1.Should().ThrowAsync<SinchApiException>();
+        }
+
+        // This test is testing a positive scenario when server was running idle - without doing requests to sinch api -
```

Comment:
See remarks from https://github.com/sinch/sinch-sdk-dotnet/pull/79#discussion_r1675947673, in particular about the server running idle: the fact a token expires has no link with the server activity.
Maybe a reference with the ticket number will help the other team members to follow what this PR is about.